src/components/display/HtmlDisplay.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useId, useState } from "react";
+import { Spin } from "antd";
+
+import type { BlobDisplayProps } from "./types";
+import { LoadingOutlined } from "@ant-design/icons";
+
+const HtmlDisplay = ({ contents, loading }: BlobDisplayProps) => {
+  const iframeId = useId();
```

Comment:
Not sure if it is worth using useRef instead?

Reply:
good point